src/Execution/Executor.php

```diff
@@ -507,21 +339,137 @@ public function completeValueWithLocatedError(
         }
     }
 
+    /**
+     * Implements the "Evaluating selection sets" section of the spec for "read" mode.
+     *
+     * @param ObjectType $objectType
+     * @param mixed      $rootValue
+     * @param array      $path
+     * @param array      $fields
+     * @return array
+     * @throws \Throwable
+     */
+    protected function executeFields(
+        ObjectType $objectType,
+        $rootValue,
+        array $path,
+        array $fields
+    ): array {
+        $finalResults       = [];
+        $doesContainPromise = false;
+
+        foreach ($fields as $fieldName => $fieldNodes) {
+            $fieldPath   = $path;
+            $fieldPath[] = $fieldName;
+
+            try {
+                $result = $this->resolveField($objectType, $rootValue, $fieldNodes, $fieldPath);
+            } catch (UndefinedException $ex) {
+                continue;
+            }
+
+            $doesContainPromise = $doesContainPromise || $this->isPromise($result);
+
+            $finalResults[$fieldName] = $result;
+        }
+
+        if ($doesContainPromise) {
+            $keys    = array_keys($finalResults);
+            $promise = \React\Promise\all(array_values($finalResults));
```

Comment:
@crisu83 If you want to remove all \React\Promise namespace, we can probably make an alias for this.

`using \React\Promise\all as promiseAll`

Reply:
Fixed.